addons/account/models/company.py

```diff
@@ -227,13 +227,23 @@ def reflect_code_prefix_change(self, old_code, new_code):
 
     def _validate_fiscalyear_lock(self, values):
         if values.get('fiscalyear_lock_date'):
+
             nb_draft_entries = self.env['account.move'].search([
                 ('company_id', 'in', self.ids),
                 ('state', '=', 'draft'),
                 ('date', '<=', values['fiscalyear_lock_date'])])
             if nb_draft_entries:
                 raise ValidationError(_('There are still unposted entries in the period you want to lock. You should either post or delete them.'))
 
+            has_unreconciled_statement_lines = self.env['account.bank.statement.line'].search_count([
+                ('company_id', 'in', self.ids),
+                ('is_reconciled', '=', False),
+            ])
+            if has_unreconciled_statement_lines:
+                raise ValidationError(_(
+                    "There are still unreconciled bank statements in the period you want to lock."
```

Comment:
```suggestion
                    "There are still unreconciled bank statement lines in the period you want to lock."
```

wouldn't that make more sense?